### PR TITLE
Replace toggle_button_disable controller with generic toggle_control controller

### PR DESCRIPTION
**Pull request by cyrillefr**
- add `enableIfPresent method in `toggle_control_controller.js` to handle enabling on toggle
- add testing in the corresponding spec
- replace in view previous controller with intended generic toggle-control

#### What? Why?
See #11928 
Does not close the issue, as this is a replacement for the first of the three old/tailored/legacy controllers.

The replaced controller was used only to toggle disability of the button to add a voucher at the checkout.

I have forgot to remove the previous controller and its spec though ...

#### What should we test?
Make a checkout and see if button to add voucher is disabled at first and then enabled when data is entered.
Or this trick:
`rspec spec/system/consumer/checkout/payment_spec.rb:128`  with putting a `binding.pry` to stop the test and see the page where the checkout + voucher is made. Need to alter cuprite setting too (headless: false)

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by cyrillefr** [comment]
Hello @dacook, could it be possible to rerun tests as they failed at "Install JS dependencies" ?

**Comment by sigmundpetersen** [comment]
done @cyrillefr

**Review by dacook — CHANGES_REQUESTED** [review]
Great, this looks perfect to me!

Just a couple of things, one below, and as you mentioned:
> remove the previous controller and its spec


Would you be able to add that as a new commit to this PR?

File: app/webpacker/controllers/toggle_control_controller.js
Comment (cyrillefr) [reply app/webpacker/controllers/toggle_control_controller.js]: Done :)

**Review by dacook — APPROVED** [review]
Wonderful, thank you!

**Review by mkllnk — APPROVED** [review]

**Comment by cyrillefr** [comment]
Hello @dacook ,

Could it be possible to merge this PR ?
I will have to make modifications to the same js controller file for the next PR that tackles #11928. (I am talking `app/webpacker/controllers/toggle_control_controller.js` that was modified).
I would rather avoid conflicts later :)

**Comment by dacook** [comment]
Hi @cyrillefr , sorry for the delay. As this affects the front end, we need to verify with a manual test before merging, just in case any unforeseen issues come up. We have a few items in the testing list at the moment so it may have to wait a little while.

In a case like this, and if you're confident with Git, I would suggest that you can create a new branch from this one, to start your next task. There's a small risk of conflicts if this branch will get changed after testing, but it seems unlikely in this case.

**Comment by cyrillefr** [comment]
Hello @dacook ,

Thank you for the clarification. My goal was certainly not to rush anyone :)

Did not know I could on github make a branch from another one.
I will go on the simpler way and wait for this one to be merged to put the next step.

**Comment by drummer83** [comment]
Thank you, @cyrillefr!

I can confirm that the apply button for the voucher code is still working correctly.

https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/6b01c2ed-321e-48ce-980e-3a0019843fd9

Great, I will merge this one and you can feel free to take the next step of this consolidation. :muscle: 

Thanks again! Merging! :partying_face:

**Pull request merged**
